### PR TITLE
feat: add support for lwt_ssl

### PR DESCRIPTION
**Pull request by c-cube**
I reimplemented that and then noticed #21, from which I adapted the certificate validation code. It seems to work for freenode for me.

### CODE REVIEW

**Comment by johnelse** [comment]
Cool, thanks!

File: irc-client-tls.opam [hunk 1]
```diff
@@ -17,7 +17,7 @@ depends: [
   "dune" {build}
   "irc-client"
   "lwt"
-  "tls"
+  "lwt_ssl"
```
Comment (hannesm) [inline irc-client-tls.opam]: this change looks wrong to me

Comment (c-cube) [reply irc-client-tls.opam]: good catch, thanks @hannesm. I first thought of replacing `irc-client-tls`, but in the end, why not just have both? :woman_shrugging:

**Comment by johnelse** [comment]
This does not build against the latest ocaml-ssl on opam (because `Ssl.TLSv1_3`) but it does build against ocaml-ssl HEAD, so it's going in.

**Pull request merged**